src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -399,6 +399,11 @@ def load_arguments(self, _):
     with self.argument_context('tag') as c:
         c.argument('tag_name', options_list=['--name', '-n'])
         c.argument('tag_value', options_list='--value')
+        c.argument('resource_id', options_list='--resource-id',
+                   help='The resource identifier for the tagged entity. A resource, a resource group or a subscription may be tagged.')
+        c.argument('tags', tags_type)
+        c.argument('operation', options_list='--operation',
+                   help='The update operation: options include Merge, Replace and Delete.')
```

Comment:
> options include Merge, Replace and Delete.

Is this an enum in the Model? Could we declare it as enum?

Reply:
At this moment, `operation` is just a string, and it should be an enum type in the Python SDK, As @jiasli  mentioned in the last pr https://github.com/Azure/azure-cli/pull/13429 (I previously defined `TagUpdateOperation` as an Enum in the `azure.cli.core.commands.parameters`, changes can be found https://github.com/Azure/azure-cli/pull/13429/commits/43341dd5ee6ecc867d23cdb6e389c215b30d5433)
I had a fix in the swagger to have `operation` generated as an Enum type: https://github.com/Azure/azure-rest-api-specs/pull/9828. The change has been merged, but need time to be picked up in the new python sdk;

To catch the S173 release as customers are pushing for Tags CLI support, I have two options here:
1. defined `TagUpdateOperation` as an Enum in `resource` module (create a `parameters.py` under `resource` module) instead of in `azure.cli.core.commands.parameters`, and use 
```python
class TagUpdateOperation(str, Enum):
    merge = "Merge"
    replace = "Replace"
    delete = "Delete"

c.argument('operation', arg_type=get_enum_type([item.value for item in list(TagUpdateOperation)]),
                   help='The update operation: options include Merge, Replace and Delete.')
```
2. Keep `operation` as a string for now, and add a logic in `custom.py` to handle this logic until new python SDK provides `operation` as an Enum:
```python
    if (not operation) or (operation.strip().lower() not in set(['merge', 'replace', 'delete'])):
        raise IncorrectUsageError("Operation could only be 'Merge', 'Replace', 'Delete'.")
```
Please kindly advise here. I temporarily chose option 2. I can make changes if you prefer the other one or have some other suggestions.

Reply:
I prefer option 1, we can replace the enum defined under resource directly when the new version of SDK contains this enum.

Reply:
Switched to option 1.